hibernate-core/src/main/java/org/hibernate/query/sqm/mutation/internal/MultiTableSqmMutationConverter.java

```diff
@@ -220,7 +222,14 @@ public Expression resolveSqlExpression(
 
 		pushProcessingState( restrictionProcessingState, getFromClauseIndex() );
 		try {
-			return (Predicate) sqmWhereClause.getPredicate().accept( this );
+			final Predicate accept = (Predicate) sqmWhereClause.getPredicate().accept( this );
+			if ( discriminatorPredicate != null ) {
+				Junction junction = new Junction( Junction.Nature.CONJUNCTION );
+				junction.add( accept );
+				junction.add( discriminatorPredicate );
+				return junction;
+			}
+			return accept;
```

Comment:
IMO this should use the existing `org.hibernate.sql.ast.spi.SqlAstHelper#combinePredicates` helper method.

```
return org.hibernate.sql.ast.spi.SqlAstHelper.combinePredicates(
    (Predicate) sqmWhereClause.getPredicate().accept( this ),
    discriminatorPredicate 
);
```